src/main/java/parsing/Parser.java

```diff
@@ -138,24 +177,43 @@ public NodeGraph parseCache(NodeGraph graph, File cache) {
             BufferedReader in = new BufferedReader(new InputStreamReader(new FileInputStream(cache)));
             int graphSize = Integer.parseInt(in.readLine());
             graph.getNodes().ensureCapacity(graphSize);
-            for (int i = 0; i < graphSize; i++) {
-                int length = Integer.parseInt(in.readLine());
-                int outLength = Integer.parseInt(in.readLine());
-                int[] outgoing = new int[outLength];
-                String[] tempLine = in.readLine().split("\t");
-                for (int j = 0; j < outLength; j++) {
-                    outgoing[j] = Integer.parseInt(tempLine[j]);
-                }
-                int inLength = Integer.parseInt(in.readLine());
-                int[] incoming = new int[inLength];
-                tempLine = in.readLine().split("\t");
-                for (int j = 0; j < inLength; j++) {
-                    incoming[j] = Integer.parseInt(tempLine[j]);
+
```

Comment:
Same as with the parse method above, is there any way you could reduce the complexity of this method?